core/src/main/java/de/danoeh/antennapod/core/feed/FeedItem.java

```diff
@@ -360,20 +328,25 @@ private boolean isPlaying() {
 
     @Override
     public Callable<String> loadShownotes() {
-        return () -> {
-            if (contentEncoded == null || description == null) {
-                DBReader.loadDescriptionOfFeedItem(FeedItem.this);
-            }
-            if (TextUtils.isEmpty(contentEncoded)) {
-                return description;
-            } else if (TextUtils.isEmpty(description)) {
-                return contentEncoded;
-            } else if (description.length() > 1.25 * contentEncoded.length()) {
-                return description;
-            } else {
-                return contentEncoded;
-            }
-        };
+        //doesn't process notes, ShownotesLoader is responsible for loading from database
+        //don't use this method
```

Comment:
If the method should not be used, it would be better to just remove it.

Reply:
There is interface `ShownotesProvider` with method `Callable<String> loadShownotes()`, previous logic was that all elements that can contain shownotes are a `ShownotesProvider`, but this led to situation where all data objects that contain shownotes became also `ShownotesProvider`, it goes against single responsibility rule (since providing shownotes involves calling database in some cases). `FeedItem` implements that interface and was calling DB to retrieve shownotes. We can actually remove 'ShownotesProvider' interface all together and move logic of providing shownotes to new service which will be solely responsible for that ( i already created class who partially does that `ShownotesLoader`).

Reply:
> We can actually remove 'ShownotesProvider' interface

Yeah, I think that would be better :+1:

Reply:
Do you have an idea how to get rid of it? Logic of AP is based on abstraction of created by `Playable` and `ShownotesProvider`. There are high level parts of applications that doesn't know about implementation of those interfaces. Unfortunately Playable, ShownotesProvider and ImageResource are implemented by main objects in app and act as services not as data objects.